hal/targets/cmsis/TARGET_ARM_SSG/TARGET_BEETLE/system_CMSDK_BEETLE.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2009-2016 ARM Limited. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * This file is derivative of CMSIS V5.00 system_ARMCM3.c
+ */
+
+#include "cmsis.h"
+
+/*----------------------------------------------------------------------------
+ * Define clocks
+ *----------------------------------------------------------------------------*/
+#define __XTAL (48000000UL) /* Oscillator frequency */
+
+#define __SYSTEM_CLOCK (__XTAL / 2)
+
+/*----------------------------------------------------------------------------
+ * Clock Variable definitions
+ *----------------------------------------------------------------------------*/
+/* !< System Clock Frequency (Core Clock) */
+uint32_t SystemCoreClock = __SYSTEM_CLOCK;
+
+/*----------------------------------------------------------------------------
+ * Clock functions
+ *----------------------------------------------------------------------------*/
+/**
+ * Update SystemCoreClock variable
+ *
+ * @param  none
+ * @return none
+ *
+ * @brief   Updates the SystemCoreClock with current core Clock
+ *          retrieved from cpu registers.
+ */
+void SystemCoreClockUpdate (void)
+{
+
+    SystemCoreClock = __SYSTEM_CLOCK;
```

Comment:
Ok, can you report it please to the cmsis team. Because the API says it should update it based on the registers read. Therefore I would expect to do this (I understand that 24MHZ is default, but a user can do a different):

```
clock  = SystemCoreClock;
// change PLL or whatever clock source in my app
UpdateSystemCoreClock();
new_clock = SystemCoreClock();
// the new clock core done, my drivers can use this value
```

We had various problem with this, that a driver does not check if clocks were updated and just used predefined value for instance .